### DISCUSSION TITLE
Can you retry the whole spec file when a test fails ?

### DISCUSSION THREAD

**Discussion by nezaba** [cypress-io/cypress#22794] [General]
Hey, im using cypress with jenkins and we have some specs that has tests with relation to other tests in the same spec. So when a single test fails, if it depends on the other test that passed before, it only retry that test so it fails anyways. Is there a way to retry the whole spec  when any test fails instead of retrying only the failed test? thanks in advance

**Comment by jjhelguero** [comment]
You should refrain from making tests dependent on previous steps. One failure should not equal all other tests failing. If you want to continue with your approach, you can use test retries at a describe level.
https://docs.cypress.io/guides/guides/test-retries#Custom-Configurations

**Reply by nezaba** [reply]
Isn't that the same result if i already specified "retries": 6 in the cypress config? The test suites config in custom configurations that you sent me says "// The per-suite configuration is applied to each test // If a test fails, it will be retried". If i understand it right, it only make the retries applicable to all the test cases in that test suite, but only retries the failed test cases instead of the whole test suit.

**Reply by jjhelguero** [reply]
Ah, that I did not know. You should try to separate your tests to not be reliant on previous tests.
https://docs.cypress.io/guides/references/best-practices#Having-tests-rely-on-the-state-of-previous-tests

**Reply by nezaba** [reply]
Yeah, this was just a really big spec file done by a coworker so i was trying to find a workaround so i dont have to rewrite it entirely, but i guess its the only way, thanks for the help!

**Reply by Chinmytosh** [reply]
Yes you're right

**Comment by Chinmytosh** [comment]
Yes you can but it's not sure that you archive that what you want.

**Comment by ghiscoding** [comment]
I was also looking for this feature and found this open issue #17321 
Yes I know that it's not recommended to have serial tests but that's my preference so please don't judge 😝
